vscode-extension/src/util.ts

```diff
@@ -301,3 +301,22 @@ export async function getPythonPath(): Promise<string> {
     pythonExtension.exports.settings.getExecutionDetails().execCommand[0];
   return pythonPath;
 }
+
+/**
+ * Checks if the specified Python interpreter is version 3.10 or higher.
+ * @param pythonPath The path to the Python interpreter.
+ * @returns A promise that resolves to a boolean indicating if the version is >= 3.10.
+ */
+
+export function isPythonVersionAtLeast310(pythonPath: string): boolean {
```

Comment:
I think this is ok to unblock but ideally if there's some way we can determine the required python version dynamically via pip or something that would be best

Reply:
what do you mean by dynamically? also pip itself has versions, which is not the same as python's version

Reply:
Oh, what I mean is our pyproject.toml defines `requires-python = ">=3.10"` and that shows up as metadata in pip. Ideally we could dynamically get that version in the code without having to hardcode it